### PR TITLE
fix: goreleaser develop build for local and CI

### PR DESCRIPTION
**Pull request by momentmaker**

### CODE REVIEW

**Review by chainchad — DISMISSED** [review]

File: core/chainlink.Dockerfile [hunk 1]
```diff
@@ -38,13 +38,14 @@ FROM ubuntu:20.04
 
 ARG CHAINLINK_USER=root
 ENV DEBIAN_FRONTEND noninteractive
-RUN apt-get update && apt-get install -y ca-certificates gnupg lsb-release curl
+RUN apt-get update && apt-get install -y ca-certificates gnupg lsb-release curl patchelf
```
Comment (erikburt) [inline core/chainlink.Dockerfile:41]: Should patchelf only be installed on the goreleaser built docker images?

File: core/chainlink.devspace.Dockerfile [hunk 2]
```diff
@@ -38,7 +38,7 @@ FROM golang:1.21-bullseye
 
 ARG CHAINLINK_USER=chainlink
 ENV DEBIAN_FRONTEND noninteractive
-RUN apt-get update && apt-get install -y ca-certificates gnupg lsb-release curl
+RUN apt-get update && apt-get install -y ca-certificates gnupg lsb-release curl patchelf
```
Comment (erikburt) [inline core/chainlink.devspace.Dockerfile:41]: Should patchelf only be installed on the goreleaser built docker images?

File: core/chainlink.goreleaser.Dockerfile [hunk 3]
```diff
@@ -6,17 +6,21 @@ FROM ubuntu:20.04
...
+  && apt-get clean all \
+  && rm -rf /var/lib/apt/lists/*
 
-COPY . /usr/local/bin/
+COPY ./chainlink /usr/local/bin/
 # Copy native libs if cgo is enabled
 COPY ./tmp/linux_${TARGETARCH}/libs /usr/local/bin/libs
+COPY ./tools/bin/ldd_fix /usr/local/bin/ldd_fix
+RUN chmod +x /usr/local/bin/ldd_fix
+RUN /usr/local/bin/ldd_fix
```
Comment (erikburt) [inline core/chainlink.goreleaser.Dockerfile:23]: Should we uninstall `patchelf` once we've finished patching the binary?

File: tools/bin/ldd_fix [hunk 4]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
```
Comment (erikburt) [inline tools/bin/ldd_fix:2]: I'd maybe add a comment to this file explaining why it's needed at all, just for future reference.

File: core/chainlink.Dockerfile (hunk 1 above)
Comment (momentmaker) [reply core/chainlink.Dockerfile:41]: yeah good catch! the regular Dockerfile doesn't need this.

File: core/chainlink.devspace.Dockerfile (hunk 2 above)
Comment (momentmaker) [reply core/chainlink.devspace.Dockerfile:41]: Sergey mentioned his the slack thread that he'd like this - will double-check with him.

**Review by erikburt — DISMISSED** [review]

File: core/scripts/go.mod [hunk 5]
```diff
@@ -1,6 +1,6 @@
 module github.com/smartcontractkit/chainlink/core/scripts
 
-go 1.21.3
+go 1.21.5
```
Comment (jmank88) [inline core/scripts/go.mod:3]: We're up to 1.21.7 now
```suggestion
go 1.21.7
```

File: .github/actions/goreleaser-build-sign-publish/action_utils [hunk 6]
```diff
@@ -71,8 +72,13 @@ goreleaser_release() {
     rm -rf cosign.key
   fi
 
+  delimiter="$(openssl rand -hex 8)"
+  echo "metadata<<${delimiter}" >> "$GITHUB_OUTPUT"
   echo "metadata=$(cat dist/metadata.json)" >> "$GITHUB_OUTPUT"
+  echo "${delimiter}" >> "$GITHUB_OUTPUT"
+  echo "artifacts<<${delimiter}" >> "$GITHUB_OUTPUT"
   echo "artifacts=$(cat dist/artifacts.json)" >> "$GITHUB_OUTPUT"
+  echo "${delimiter}" >> "$GITHUB_OUTPUT"
```
Comment (chainchad) [inline .github/actions/goreleaser-build-sign-publish/action_utils:81]: What is this accomplishing?

File: go.mod [hunk 7]
```diff
@@ -1,6 +1,6 @@
 module github.com/smartcontractkit/chainlink/v2
 
-go 1.21.3
+go 1.21.5
```
Comment (chainchad) [inline go.mod:3]: Any reason this is a different vers than `core/scripts/go.mod`?

File: integration-tests/go.mod [hunk 8]
```diff
@@ -1,6 +1,6 @@
 module github.com/smartcontractkit/chainlink/integration-tests
 
-go 1.21.4
+go 1.21.5
```
Comment (chainchad) [inline integration-tests/go.mod:3]: Different than `core/scripts/go.mod`, by design?

File: tools/bin/ldd_fix [hunk 9]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
```
Comment (chainchad) [inline tools/bin/ldd_fix:1]:
```suggestion
#!/usr/bin/env bash
```

File: tools/bin/ldd_fix [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+if [ -z "$path" ]; then
+  echo "Error: Failed to extract the path from the line."
+  exit 1
+fi
+
+trimmed_path=${path%.so*}.so
+cosm_file=$(ls ${libs_path} | grep "\.so$" | head -n 1)
+
+patchelf --remove-needed "${trimmed_path}" "$chainlink_path"
+patchelf --add-needed "$cosm_file" "$chainlink_path"
```
Comment (chainchad) [inline tools/bin/ldd_fix:27]: 🪄

File: go.mod (hunk 7 above)
Comment (momentmaker) [reply go.mod:3]: good question - @jmank88 are all of them now `1.21.7`?

File: .github/actions/goreleaser-build-sign-publish/action_utils (hunk 6 above)
Comment (chainchad) [reply .github/actions/goreleaser-build-sign-publish/action_utils:81]: Still a little confused about the delimiter. Maybe we can add some comments?

**Review by chainchad — APPROVED** [review]

**Pull request merged**